### PR TITLE
Remove ".paper-remapped/" from plugin path

### PR DESCRIPTION
**Pull request by pavog**
In the latest version 1.21, Paper has changed some internals so that the plugins are apparently loaded from the path `/plugins/.paper-remapped/` instead of `/plugins/`. This also causes the error messages to change the plugin path from `/plugins/<filename>` to `/plugins/.paper-remapped/<filename>`.

We extract the plugin path from the log and use it in the problem message that we display to the user.
We also use the plugin path to display it in solutions.  E.g.: _Delete the file `plugins/.paper-remapped/mclogs-bukkit-dev.jar`._
But the user cannot see or access the .paper-remapped folder, so they cannot delete the file from there.
**This can lead to confusion.**
Therefore, this PR changes this behavior and replaces the path `/plugins/.paper-remapped/` with `/plugins/` in all places.

In addition, this PR changes the following:
- Two new functions have been added to the `PluginProblem` class :
   - `correctPluginPath` changes `plugins/.paper-remapped/` to `plugins/`
   - `extractPluginName` can extract the name of the plugin from the path, this can simplify some regexes in the problem classes
- :warning: Many of the problems that are actually plugin problems now extend from the PluginProblem class, so the member variable `$pluginName` can be removed and they offer the `PluginInstallDifferentVersionSolution` 
- Adds some new tests for PaperMC 1.21.1
- Adds the `PluginInstallDifferentVersionSolution` to `UnsupportedClassVersionProblem`

### CODE REVIEW

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 1]
```diff
@@ -102,14 +83,13 @@ public static function getPatterns(): array
...
-        $this->pluginName = $matches[2];
-        $this->dependencyPluginNames = preg_split("/, ?/", $matches[3]);
+        $this->pluginName = $this->extractPluginName($matches[1]);
+        $this->dependencyPluginNames = preg_split("/, ?/", $matches[2]);
 
         foreach ($this->dependencyPluginNames as $name) {
             $this->addSolution((new PluginInstallSolution())->setPluginName($name));
         }
-        $this->addSolution((new FileDeleteSolution())->setRelativePath($this->getPluginPath()));
+        $this->addSolution((new PluginRemoveSolution())->setPluginName($this->getPluginName()));
```
Comment (JulianVennen) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:92]: I think we should prefer providing the exact path of the plugin that should be deleted over only showing the plugin name.

**Review by JulianVennen — CHANGES_REQUESTED** [review]

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php (hunk 1 above)
Comment (pavog) [reply src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php]: I added `BukkitPluginFileProblem` that stores the plugins file path and passes it to the `PluginRemoveFileSolution`.

**Review by JulianVennen — APPROVED** [review]

File: test/data/Vanilla/Bukkit/Folia/folia-1-19-4-unsupported-plugin.json [hunk 2]
```diff
@@ -1971,7 +1971,7 @@
     "analysis": {
         "problems": [
             {
-                "message": "The plugin motdgg does not support regional ticking.",
+                "message": "The plugin motdgg-bukkit-1.2.0 does not support regional ticking.",
```
Comment (matthi4s) [inline test/data/Vanilla/Bukkit/Folia/folia-1-19-4-unsupported-plugin.json:1974]: This should still load the plugin name and not the file name.

File: test/data/Vanilla/Bukkit/Paper/paper-multiple-depepdencies-1-21-1.log [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[12:47:34] [ServerMain/INFO]: [bootstrap] Running Java 21 (OpenJDK 64-Bit Server VM 21.0.4+7-LTS; Eclipse Adoptium Temurin-21.0.4+7) on Linux 5.15.0-117-generic (amd64)
```
Comment (matthi4s) [inline test/data/Vanilla/Bukkit/Paper/paper-multiple-depepdencies-1-21-1.log:1]: There is a typo in the file name

File: test/data/Vanilla/Bukkit/Paper/paper-plugin-unsupported-class-version-1-16-5.json [hunk 4]
```diff
@@ -464,7 +464,7 @@
     "analysis": {
         "problems": [
             {
-                "message": "The plugin 'Slimefun4-1027' requires Java version 16 or later to run.",
+                "message": "The plugin 'Slimefun4-1027' requires Java version -44 or later to run.",
```
Comment (matthi4s) [inline test/data/Vanilla/Bukkit/Paper/paper-plugin-unsupported-class-version-1-16-5.json:467]: Looks like the class version wasn't read correctly

**Review by matthi4s — CHANGES_REQUESTED** [review]

File: test/data/Vanilla/Bukkit/Paper/paper-multiple-depepdencies-1-21-1.log (hunk 3 above)
Comment (pavog) [reply test/data/Vanilla/Bukkit/Paper/paper-multiple-depepdencies-1-21-1.log]: Fixed

File: test/data/Vanilla/Bukkit/Paper/paper-plugin-unsupported-class-version-1-16-5.json (hunk 4 above)
Comment (pavog) [reply test/data/Vanilla/Bukkit/Paper/paper-plugin-unsupported-class-version-1-16-5.json]: Fixed

File: test/data/Vanilla/Bukkit/Folia/folia-1-19-4-unsupported-plugin.json (hunk 2 above)
Comment (pavog) [reply test/data/Vanilla/Bukkit/Folia/folia-1-19-4-unsupported-plugin.json]: Fixed

**Review by matthi4s — APPROVED** [review]

**Review by JulianVennen — APPROVED** [review]

**Review by matthi4s — APPROVED** [review]

**Pull request merged**